akvo/rsr/views/py_reports/program_period_labels_overview_pdf_report.py

```diff
@@ -216,35 +241,54 @@ def progress(self):
         return calculate_percentage(self.actual_value, self.target_value)
 
     def add_value(self, value, disaggregations={}):
-        return self._replace(
+        return replace(
+            self,
             value=ensure_decimal(self.value) + ensure_decimal(value),
             disaggregations=self._add_disaggregation(disaggregations)
         )
 
     def add_percentage_value(self, numerator, denominator, disaggregations={}):
-        return self._replace(
+        return replace(
+            self,
             numerator=ensure_decimal(self.numerator) + ensure_decimal(numerator),
             denominator=ensure_decimal(self.denominator) + ensure_decimal(denominator),
             disaggregations=self._add_disaggregation(disaggregations)
         )
 
     def get_disaggregation_of(self, category, type):
         key = (category, type)
-        if key not in self.disaggregations:
+        if key not in self.disaggregations and key not in self.period_disaggregations:
             return None
+        if self.cumulative:
+            return self.period_disaggregations.get(key, {'value': None})['value']
         if self.type == IndicatorType.UNIT:
             return self.disaggregations[key]['value']
         d = self.disaggregations[key]
         return calculate_percentage(d['numerator'], d['denominator'])
 
-    def _add_disaggregation(self, new):
+    def _add_disaggregation(self, disaggregations):
         result = copy.deepcopy(self.disaggregations)
-        for key in new:
+        for key, disaggregation in disaggregations.items():
             if key not in result:
-                result[key] = new[key].copy()
+                result[key] = disaggregation.copy()
             elif self.type == IndicatorType.UNIT:
-                result[key]['value'] += new[key]['value']
+                result[key]['value'] += disaggregation['value']
             else:
-                result[key]['numerator'] += new[key]['numerator']
-                result[key]['denominator'] += new[key]['denominator']
+                result[key]['numerator'] += disaggregation['numerator']
+                result[key]['denominator'] += disaggregation['denominator']
         return result
+
+    @cached_property
+    def period_disaggregations(self):
+        disaggregations = {}
+        for d in self.period.disaggregations.all():
+            category = d.dimension_value.name.name
+            type = d.dimension_value.value
+            disaggregations[(category, type)] = {
+                'category': category,
+                'type': type,
+                'value': d.value,
+                'numerator': d.numerator,
+                'denominator': d.denominator,
+            }
```

Comment:
`DisaggregationData` might be useful here instead of a dict

Reply:
These changes will take longer. Better if done in a separate PR.